### PR TITLE
Remove deprecated classes from app-model

### PR DESCRIPTION
**Pull request by gastaldi**
I am not sure if it's too early to remove them but at least it would be good to have it gone in the next LTS

### CODE REVIEW

**Review by aloubyansky — APPROVED** [review]

**Comment by rsvoboda** [comment]
@gastaldi @aloubyansky just checking about RHBQ implications as the deprecation was done in 3.11 (thus not in the latest RHBQ 3.8) through https://github.com/quarkusio/quarkus/pull/40224.

Are these classes in scope of any support level from RHBQ side?

File: core/deployment/src/main/java/io/quarkus/deployment/ApplicationArchive.java [hunk 1]
```diff
@@ -72,12 +71,6 @@ public interface ApplicationArchive {
      */
     PathCollection getResolvedPaths();
 
-    /**
-     * @deprecated in favor of {@link #getKey()}
-     * @return the artifact key or null if not available
-     */
-    AppArtifactKey getArtifactKey();
```
Comment (aloubyansky) [inline core/deployment/src/main/java/io/quarkus/deployment/ApplicationArchive.java:79]: This method was deprecated in 2.4.0.CR1

**Review by aloubyansky — COMMENTED** [review]
The methods that exposed the removed API seem to be deprecated a long time ago (2.4.0.CR1), so those could be removed.
We could still keep the `AppArtifact*` classes around but they wouldn't be used in the quarkus repository at least. Theoretically, there could be uses of those in other extension repositories, which would be easy to fix.
If we want to be strict about this, we could add the classes back, we could also wait until somebody complains before doing that.

**Comment by gsmet** [comment]
Sorry to be that guy but we need the deprecation to land in an LTS before a removal.
So I would reintroduce the classes.
Even if we don't have a public extension using them, we could have a private one.

**Comment by gastaldi** [comment]
These methods were deprecated since 2.x, we had 2 LTS with them being deprecated already, so I don't understand why we need to reintroduce them?

**Comment by rsvoboda** [comment]
My comment was about `App*` classes deprecated in 3.11 through https://github.com/quarkusio/quarkus/pull/40224

**Comment by gastaldi** [comment]
How about if we introduced an OpenRewrite script to migrate its usage? Anyway, not opposed to it, just considering options

**Comment by gsmet** [comment]
We can introduce an OpenRewrite recipe, that's indeed a good idea if feasible.

But that doesn't change the rule :).

**Comment by aloubyansky** [comment]
That would be mostly for extension code though, not applications.

**Comment by gsmet** [comment]
The recipes also handle updating extensions.

**Comment by gastaldi** [comment]
@gsmet PR created in https://github.com/quarkusio/quarkus/pull/42052

**Pull request merged**